lib/registry.py

```diff
@@ -365,10 +365,13 @@ def headers_log(hs):
          ch.INFO("Docker Hub rate limit: %s pulls left of %s per %s hours (%s)"
                  % (left_ct, pull_ct, period, reason))
 
+   @property
+   def _url_base(self):
+      return "https://%s:%d/v2/" % (self.ref.host, self.ref.port)
+   
    def _url_of(self, type_, address):
       "Return an appropriate repository URL."
-      url_base = "https://%s:%d/v2" % (self.ref.host, self.ref.port)
-      return "/".join((url_base, self.ref.path_full, type_, address))
+      return self._url_base + "/".join((self.ref.path_full, type_, address))
```

Comment:
This will add a redundant slash, no?

Reply:
I removed the base url from the `join()` method in attempt to avoid a redundant slash. Here's an example of what's going on in line 374
```
>>> "foo/" + "/".join(["bar", "baz"])
'foo/bar/baz
```

Reply:
Oh sorry I'm just wrong. Never mind.